compiler/rustc_borrowck/src/diagnostics/mod.rs

```diff
@@ -623,6 +623,28 @@ impl UseSpans<'_> {
         }
     }
 
+    // callback a subdiag on the label-span
+    pub(super) fn var_span_subdiag_label(
+        self,
+        err: &mut Diagnostic,
+        cb: impl Fn(Span) -> crate::session_diagnostics::CaptureVarCause,
+        kind_desc: impl Into<String>,
+    ) {
+        if let UseSpans::ClosureUse { capture_kind_span, path_span, .. } = self {
+            let (var_span, some_kind_span) = if capture_kind_span == path_span {
+                (capture_kind_span, None)
+            } else {
+                (path_span, Some(capture_kind_span))
+            };
+            if let Some(kind_span) = some_kind_span {
+                err.subdiagnostic(crate::session_diagnostics::CaptureVarKind::Here {
+                    kind_desc: kind_desc.into(),
+                    kind_span,
+                });
+            }
+            err.subdiagnostic(cb(var_span));
+        }
```

Comment:
```suggestion
        if let UseSpans::ClosureUse { capture_kind_span, path_span, .. } = self {
            if capture_kind_span == path_span {
				f(capture_kind_span);
            } else {
                err.subdiagnostic(crate::session_diagnostics::CaptureVarKind::Here {
                    kind_desc: kind_desc.into(),
                    capture_kind_span,
                });
                f(path_span);
            }
        }
```
I think this is a little more readable if structured like the original function, something like the above.